voip_utils/sip.py

```diff
@@ -283,3 +353,277 @@ def _parse_sip(
+                            # We should probably tell the associated RTP server to shutdown at this point, assuming we aren't reusing it for other calls
+                            _LOGGER.debug("Sending OK for BYE message: %s", ok_text)
+                            self.transport.sendto(
+                                ok_bytes,
+                                (self._dest_endpoint.host, self._dest_endpoint.port),
+                            )
+
+                            self.transport.close()
+                            self.transport = None
+                        else:
+                            _LOGGER.debug("Skipping message: %s", line)
+                elif not line:
+                    break
+
+            if is_ok:
+                _LOGGER.debug("Got OK message")
+                if self.transport is not None:
+                    protocol, code, reason, headers, body = self._parse_sip_reply(
+                        response_text
+                    )
+                    _LOGGER.debug(
+                        "Parsed response protocol=%s code=%s reason=%s headers=[%s] body=[%s]",
+                        protocol,
+                        code,
+                        reason,
+                        headers,
+                        body,
+                    )
+                    rtp_info = get_rtp_info(body)
+                    remote_rtp_port = rtp_info.rtp_port
+                    opus_payload_type = rtp_info.payload_type
+                    to_header = headers["to"]
+                    ack_lines = [
+                        f"ACK {self._request_uri} SIP/2.0",
+                        f"Via: SIP/2.0/UDP {self._source_endpoint.host}:{self._source_endpoint.port}",
+                        f"From: <sip:{self._source_endpoint.username}@{self._source_endpoint.host}:{self._source_endpoint.port}>",
+                        f"To: {to_header}",
+                        f"Call-ID: {self._call_id}",
+                        "CSeq: 50 ACK",
+                        "User-Agent: test-agent 1.0",
+                        "Content-Length: 0",
+                    ]
+                    ack_text = _CRLF.join(ack_lines) + _CRLF
+                    ack_bytes = ack_text.encode("utf-8")
+                    self.transport.sendto(
+                        ack_bytes, (self._dest_endpoint.host, self._dest_endpoint.port)
+                    )
+
+                    # The call been answered, proceed with desired action here
+                    self.on_call(
+                        CallInfo(
+                            caller_endpoint=self._dest_endpoint,
+                            caller_rtp_port=remote_rtp_port,
+                            server_ip=self._dest_endpoint.host,
+                            headers=headers,
+                            opus_payload_type=opus_payload_type,  # Should probably update this to eventually support more codecs
+                        )
+                    )
+        except Exception:
```

Comment:
Do we really need this? Can we specify which exceptions we expect ?

Reply:
This is probably out of scope for this exercise but the SipDatagramProtocol datagram_received method was also using a try catch, which is why I did the same here. I would suspect we probably want to remove the try/except there as well, and also change the places it was raising ValueError to make it consistent with this new class.